#24 should be repalced by however many bytes in gps data
uint8[] raw_gps_data
uint8[24] size_gps_data